xarray/core/_typed_ops.pyi

```diff
@@ -0,0 +1,731 @@
+"""Stub file for mixin classes with arithmetic operators."""
+# This file was generated using xarray.util.generate_ops. Do not edit manually.
+
+from typing import NoReturn, TypeVar, Union, overload
+
+import numpy as np
+
+from .dataarray import DataArray
+from .dataset import Dataset
+from .groupby import DataArrayGroupBy, DatasetGroupBy, GroupBy
+from .variable import Variable
+
+try:
+    from dask.array import Array as DaskArray
+except ImportError:
+    DaskArray = np.ndarray
+
+T_Dataset = TypeVar("T_Dataset", bound="DatasetOpsMixin")
+T_DataArray = TypeVar("T_DataArray", bound="DataArrayOpsMixin")
+T_Variable = TypeVar("T_Variable", bound="VariableOpsMixin")
+
+Scalar = Union[int, float, complex, bytes, str, np.generic]
+ArrayLike = Union[np.ndarray, DaskArray]
+ScalarOrArray = Union[Scalar, ArrayLike]
```

Comment:
The following works 
```python
da = xr.DataArray([1, 2, 3])
a + [1, 2, 3]
```
so I wonder if you have to add `ArrayLike` from `np.typing` here (need to add it to `core/npcompat.py`)

Reply:
Great idea! I have now added it to `npcompat.py` including a fallback for installation with older numpy versions. The fallback is based on code from numpy 1.20. Numpy's `ArrayLike` already includes Python scalars so that simplifies the type definitions we need at the top of the generated file.